flink-connectors/flink-connector-hbase-base/src/main/java/org/apache/flink/connector/hbase/options/HBaseOptionsUtil.java

```diff
@@ -0,0 +1,145 @@
+                                                + "The primary key of HBase table must be defined on row key field.");
+                            }
+                            if (!hbaseSchema.getRowKeyName().get().equals(k.getColumns().get(0))) {
+                                throw new IllegalArgumentException(
+                                        "Primary key of HBase table must be defined on the row key field. "
+                                                + "A row key field is defined as an atomic type, "
+                                                + "column families and qualifiers are defined as ROW type.");
+                            }
+                        });
+    }
+
+    public static HBaseWriteOptions getHBaseWriteOptions(ReadableConfig tableOptions) {
+        HBaseWriteOptions.Builder builder = HBaseWriteOptions.builder();
+        builder.setBufferFlushIntervalMillis(
+                tableOptions.get(SINK_BUFFER_FLUSH_INTERVAL).toMillis());
+        builder.setBufferFlushMaxRows(tableOptions.get(SINK_BUFFER_FLUSH_MAX_ROWS));
+        builder.setBufferFlushMaxSizeInBytes(
+                tableOptions.get(SINK_BUFFER_FLUSH_MAX_SIZE).getBytes());
+        builder.setParallelism(tableOptions.getOptional(SINK_PARALLELISM).orElse(null));
+        return builder.build();
+    }
+
+    public static HBaseLookupOptions getHBaseLookupOptions(ReadableConfig tableOptions) {
+        HBaseLookupOptions.Builder builder = HBaseLookupOptions.builder();
+        builder.setLookupAsync(tableOptions.get(HBaseOptions.LOOKUP_ASYNC));
+        builder.setMaxRetryTimes(tableOptions.get(HBaseOptions.LOOKUP_MAX_RETRIES));
+        builder.setCacheExpireMs(tableOptions.get(HBaseOptions.LOOKUP_CACHE_TTL).toMillis());
+        builder.setCacheMaxSize(tableOptions.get(LOOKUP_CACHE_MAX_ROWS));
+        return builder.build();
+    }
+
+    /**
+     * config HBase Configuration.
+     *
+     * @param options properties option
+     */
+    public static Configuration getHBaseConfiguration(Map<String, String> options) {
+        org.apache.flink.configuration.Configuration tableOptions =
+                org.apache.flink.configuration.Configuration.fromMap(options);
+        // create default configuration from current runtime env (`hbase-site.xml` in classpath)
+        // first,
+        Configuration hbaseClientConf = HBaseConfigurationUtil.getHBaseConfiguration();
+        hbaseClientConf.set(HConstants.ZOOKEEPER_QUORUM, tableOptions.getString(ZOOKEEPER_QUORUM));
+        hbaseClientConf.set(
+                HConstants.ZOOKEEPER_ZNODE_PARENT, tableOptions.getString(ZOOKEEPER_ZNODE_PARENT));
+        // add HBase properties
+        final Properties properties = getHBaseClientProperties(options);
+        properties.forEach((k, v) -> hbaseClientConf.set(k.toString(), v.toString()));
+        return hbaseClientConf;
+    }
+
+    private static Properties getHBaseClientProperties(Map<String, String> tableOptions) {
```

Comment:
could we add a `[hotfix]` commit to replace this prefix magic? We can replace this with a map option type nowadays. Also for the Kafka connector and maybe also for the Kinesis connector. Otherwise the new Option classes are not very useful if they are less complete.